android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/KMLog.java

```diff
@@ -45,4 +47,27 @@ public static void LogException(String tag, String msg, Throwable e) {
       Sentry.captureException(e);
     }
   }
+
+  /**
+   * Utility to include JSONArray in the log exception sent to Sentry
+   * @param tag String of the caller
+   * @param msg String of the exception message
+   * @param array JSONArray of the installed keyboards_list.json file
+   * @param e Throwable exception
+   */
+  public static void LogJSONException(String tag, String msg, JSONArray array, Throwable e) {
```

Comment:
While the name of this function is generalised, what it actually does is closely tied to `keyboard_list`. I think we should probably simplify:

1. We don't need to try and pretty print it -- that's easy for us to do by hand when researching the issue (and as you note in the code, it may fail anyway)
2. The extra variable should be called `json` rather than `keyboards_list.json`. 
3. A thought: can we take in a more general object: accept any `Object` (as `JSONArray` sadly inherits from `Object` and not a generic JSON base type) and use `toString()` on that? In this case, maybe this should be a more generic `LogExceptionWithData` or something like that?
4. Don't forget to wrap the `setExtra` call with an exception handler so that if it fails, we still capture the original error

Reply:
I've changed the logger to `LogExceptionWithData()` which takes an object name and the object to print.